src/EFCore.PG/Query/Sql/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -202,31 +202,52 @@ protected override Expression VisitUnary(UnaryExpression expression)
             return base.VisitUnary(expression);
         }
 
+        /// <inheritdoc />
+        protected override Expression VisitIndex(IndexExpression expression)
+        {
+            // text cannot be subscripted.
+            if (expression.Object.Type == typeof(string))
+            {
+                return VisitSqlFunction(
+                    new SqlFunctionExpression(
+                        "substr",
+                        typeof(char),
+                        new[] { expression.Object, GenerateOneBasedIndexExpression(expression.Arguments[0]), Expression.Constant(1) }));
+            }
+
+            Visit(expression.Object);
```

Comment:
I don't think we're sure that we're dealing with an array at this point... I think a check is missing (immediately return null otherwise)